### ISSUE TITLE
Warrior gcd's feel slow

### ISSUE THREAD

**Issue by nezterx** [gamefreedomgit/Maelstrom#3761]
[//]: # (REMBEMBER! Add links to things related to the bug using for example:)
[//]: # (http://wowhead.com/)
[//]: # (cata-twinhead.twinstar.cz)

**Description:**
MS - Mortal Strike
CS - Colossus Smash
OP - Overpower
I will be using abbreviations to make my life easier writing up this post, hopefully it makes somewhat sense.

The gcd interaction with arms warrior seems slow or laggy compared to other Cataclysm servers. As a result, it makes it extremely inconsistent to reliably push four gcd's into every colossus smash and the natural flow of the spec is lagging behind further resulting in lost overpower procs due to the off-set caused with the gcd mechanic.

https://www.youtube.com/watch?v=2yYrZqDSLig
- timestamps of successfully 4 abilities in a single CS window: 2:34, 2:57, 3:31 (IF he ended the ability sequence on an overpower there, rather than a slam), 4:06 and 4:38 (IF he opted for using Execute or overpower instead of waiting for MS to come off cd).

https://www.youtube.com/watch?v=mSKwUvk87zQ
- timestamps of successfully 4 abilities in a single CS window: 0:25, 1:18, 1:57, 4:02, 4:18

https://www.twitch.tv/videos/846689605?collection=mEkxkIZKNxazkA
- Almost anywhere in the video you will see me consistently push 4 gcd's into CS.
- First 25 seconds of the fight you can also see what I mean about the link between having clean gcd's and not overlapping Taste of Blood procs.

https://www.twitch.tv/videos/846689605?collection=mEkxkIZKNxazkA
- older video but sorta shows the same stuff as the other one

To verify if the 4th ability actually registered under the CS effect then just find a different timestamp in the video and compare
hits/crits out of CS against hits/crits inside CS (not the 4th ability). Enrage/Landslide/Pots are the variables but they are insignificant in comparison to CS when it comes to identifying if the 4th ability in the CS window was registered under the effect or not.

To give an example, here on Whitemane you would always experience Taste of Blood proc overlaps by using these variations:
CS -> MS -> Slam -> Slam -> OP
CS -> MS -> Slam -> OP -> OP
CS -> MS -> OP -> Slam -> OP
CS -> Slam -> MS -> OP
CS -> MS -> Slam/OP -> OP/Slam -> MS - this one being the "slowest" variation to execute and yet still providing retail Cata video proving that this 4 gcd variation should all fit into a single CS.
(p.s.), there are way more variations, OP and Slam can be used interchangeably in most variations, as long as you don't end the combo on Slam.

There are other variations but in most simplistic terms, one of those combos would be used under different situations and that would be decided on the status of Taste of Blood proc.
On Whitemane it's very rare to successfully be able to pull off 4 abilities in a single CS. As showcased in my own vods from Twinstar servers and arms warrior videos from retail cata, this doesn't look nor feel right here on Whitemane.

**How to reproduce:**
A good way to reproduce and see if you can fit 4 abilities into CS yourself is to try following combos
CS -> MS -> Slam -> Slam -> OP
and
CS -> Slam -> MS -> OP

On Whitemane you will rarely see that final OP to make the cut inside the CS window.

**How it should work:**
You should be able to reliably push 4 gcd's into every CS window.

**Database links:**
https://cata-twinhead.twinstar.cz/?spell=86346

**Comment by UltraNix** [comment]
@nezterx What is your haste rating?

**Comment by bonfax4** [comment]
> 1000ms min gcd is a global variable to all spells. Melee spells gcd (as I know) should not be afftected by haste.

Yeah you're right. Thought something changed in cata but not really. Haste should not affect melee abilities. Could have smthing to do with debuff application delays too i guess ? I know retail had slight ms delays to those which we prob dont have here. I can see warrior fitting 4 executes easily in colossus smash window [here](https://youtu.be/2yYrZqDSLig)

**Comment by UltraNix** [comment]
As I can see the problem is that you cannot queue a new spell if there is still active cd on it. That's why you cannot cast another MS before CS is gone. Queuing spell on gcd is okay, but queueing spell on cd is soemthing broken (client blocks it). 
You can test it with any spell with cooldown - you will be not allowed to queue the spell until the cd is off.

**Comment by Warlorc** [comment]
> 1000ms min gcd is a global variable to all spells. Melee spells gcd (as I know) should not be afftected by haste.

https://wowpedia.fandom.com/wiki/Cooldown

mentions only energy bases classes (cause they already had a built in reduction) are not affected by haste to reduce gcd.

**Comment by Warlorc** [comment]
"The base global cooldown is generally 1.5 seconds for all classes except rogues, Ability druid (catform), and monks, whose abilities mostly have one second global cooldowns. However, since Patch 2.4.0, haste can reduce the global cooldown to a minimum of 0.75 seconds (1.0 seconds prior to Legion) for the spells that originated from 1.5 second cooldown. Haste doesn't affect GCD of agility energy melee classes like the ones mentioned before. A few specific abilities have global cooldowns as short as 0.5 seconds. The global cooldown affects the wait for the next ability, so using an item or ability with the standard 1.5 second cooldown will require waiting that long before a 1 second global cooldown ability can be used."

**Comment by bonfax4** [comment]
https://wowpedia.fandom.com/wiki/Haste?oldid=2915257#Spell_global_cooldown 

Very specific on abils classified as spells only if you go back up to 2012. Few [mmo posts ](https://www.mmo-champion.com/threads/1084675-Does-haste-lower-GCD) mentioning this too

 I see no difference in gcd also before and after bloodlust on old cata videos on warriors. 30% haste would make warriors close if not gcd capped otherwise.
